pkg/tarmak/tarmak.go

```diff
@@ -365,3 +374,135 @@ func (t *Tarmak) CmdKubectl(args []string) error {
 func (t *Tarmak) CancellationContext() interfaces.CancellationContext {
 	return t.ctx
 }
+
+func (t *Tarmak) DestroyEnvironment() error {
+	inputDestroy := input.New(os.Stdin, os.Stdout)
+
+	destroyClusters := true
+	moveFolder := true
+	removeConfig := true
+
+	if !t.flags.Environment.Destroy.AutoApprove {
+		d, err := inputDestroy.AskYesNo(&input.AskYesNo{
+			Default: false,
+			Query:   "Destroy all clusters?",
+		})
+		if err != nil {
+			return err
+		}
+		destroyClusters = d
+	}
+
+	if destroyClusters {
+		t.log.Info("Destroying clusters")
+		for _, cluster := range t.Environment().Clusters() {
+			// We first want to destroy the k8s clusters before the hub
+			if cluster.Name() != t.Environment().Hub().Name() {
+				t.cluster = cluster
+				if err := t.DestroyActivecluster(); err != nil {
+					return err
+				}
+			}
+		}
+
+		// After destroying all other clusters, we can destroy the hub
+		t.cluster = t.Environment().Hub()
+		if err := t.DestroyActivecluster(); err != nil {
+			return err
+		}
+	} else {
+		for _, cluster := range t.Environment().Clusters() {
+			hosts, err := cluster.ListHosts()
+			if err != nil {
+				return err
+			}
+			if len(hosts) > 0 {
+				return fmt.Errorf("can't proceed with destroying this environment, because it still has hosts running")
+			}
+		}
+	}
```

Comment:
Though I like that it is asking at every step if the user wishes to proceed, I would expect that a no answer would immediately return